airflow/jobs/scheduler_job.py

```diff
@@ -1658,6 +1658,9 @@ def _schedule_dag_run(self, dag_run: DagRun, currently_active_runs: int, session
                 )
                 return 0
 
+        # This DagRun will be schedule by this job so we set up the creating_job_id
+        dag_run.creating_job_id = self.id
```

Comment:
I think we should only set this when actually _creating_ the DagRun, not every time we schedule it, so this should be in the `_create_dag_runs` function instead.